hypothesis-python/tests/cover/test_float_nastiness.py

```diff
@@ -186,15 +186,16 @@ def test_floats_in_tiny_interval_within_bounds(data, center):
     lo = Decimal.from_float(next_down(center)).next_plus()
     hi = Decimal.from_float(next_up(center)).next_minus()
     assert float(lo) < lo < center < hi < float(hi)
-    val = data.draw(st.floats(lo, hi))
-    assert lo < val < hi
+    with pytest.raises(HypothesisDeprecationWarning):
```

Comment:
Use `pytest.warns` here, and keep the assertion outside the scope of the `with`.